### PR TITLE
chore: add predev script to check node version and env file

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

Check for correct node version and env file before running the dev command 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [ ] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by BlankParticle** [comment]
* **#772** <a href="https://app.graphite.dev/github/pr/un/inbox/772?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BlankParticle and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by McPizza0 — APPROVED** [review]

File: predev.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// Check Node version
+const nodeVersion = process.version;
+const requiredVersion = readFileSync('.nvmrc', 'utf-8').trim();
+
+if (!nodeVersion.startsWith(requiredVersion)) {
+  console.error(
+    `You are using Node ${nodeVersion}, but this project requires Node ${requiredVersion}.\nUse the correct node version to run this project`
+  );
+  process.exit(1);
+}
```
[the remark this answers is not in the document]
Comment (BlankParticle) [reply predev.ts:12]: allowing higher might cause issues, for example if I try to run the app with node 22, I get bunch of warnings related to some module that is deprecated.
We should do exact minor version match

File: predev.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+// Check for env file
+const envFile = lstatSync('.env.local', { throwIfNoEntry: false });
+if (!envFile?.isFile()) {
+  console.error(
+    'You are missing a .env.local file. Please refer to the README for instructions on how to create one.'
+  );
+  process.exit(1);
+}
```
[the remark this answers is not in the document]
Comment (BlankParticle) [reply predev.ts:21]: Ok that works

File: package.json [hunk 3]
```diff
@@ -55,7 +55,8 @@
     "ee:check": "infisical run --env=remote -- turbo run ee:check",
     "ee:build:all": "dotenv -e .env.local -- turbo run ee:build --cache-dir=.turbo",
     "mock:incoming-mail": "dotenv -e .env.local -- pnpm --dir apps/mail-bridge mock:incoming-mail",
-    "prepare": "husky"
+    "prepare": "husky",
+    "predev": "tsx predev.ts"
```
[the remark this answers is not in the document]
Comment (BlankParticle) [reply package.json:59]: pnpm automatically picks up pre scripts, so it would always run before `pnpm dev`

**Comment by BlankParticle** [comment]
### Merge activity

* **Sep 11, 8:11 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/un/inbox/772) rebased this pull request as part of a merge.

**Pull request merged**